src/DiagnosticMessages.ts

```diff
@@ -590,6 +590,11 @@ export let DiagnosticMessages = {
         message: `There are multiple components with the name '${componentName}'`,
         code: 1115,
         severity: DiagnosticSeverity.Error
+    }),
+    functionCannotHaveSameNameAsClass: (className: string) => ({
+        message: `Function cannot have the same name as class '${className}'`,
```

Comment:
Maybe more consistent with `LocalVar` message?

Reply:
@elsassph which LocalVar message are you referring to?

Reply:
Oh, this one: `localVarSameNameAsClass`. Yeah, good call. It's subtle, but they should be consistent.